### PR TITLE
Rename freedesktop and metainfo files to reverse DNS schema

### LINKED ISSUE #553: Rename files to reverse DNS schema
### Guidelines

- [X] I have read the guidelines.

### Is your feature request related to a problem? Please describe.

It's a reminder for https://github.com/tenacityteam/tenacity/pull/547#discussion_r697840022, suggesting to change e.g. `tenacity.desktop` to `org.tenacityaudio.Tenacity.desktop`.

According to @leio, "[...] regular installs should have reverse DNS notation desktop and metainfo files as well. This becomes important if we ever make use of things like dbus service activation further down the line."

### Describe the solution you'd like

Renaming everything to the reverse DNS schema counterpart, so `org.tenacityaudio.Tenacity.*`.

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### This issue is not a duplicate

- [X] I have confirmed this issue isn't a duplicate.

### PR DESCRIPTION
**Pull request by lmarz**
Signed-off-by: Leon Marz <main@lmarz.org>

Resolves: #553

Rename the files `src/tenacity.desktop.in` and `src/tenacity.xml` to the reverse DNS schema.
The mime info file `src/tenacity.xml` doesn't really have a file naming convention but I also renamed it for consistency within the freedesktop.org ecosystem.
@TheEvilSkeleton Are there more files that need to be renamed?

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by n0toose** [comment]
Seems fine to me, but leaving the review up to @TheEvilSkeleton.

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]
Can you change the path https://github.com/tenacityteam/tenacity/blob/e776c2f7e41cad437edf6385fa42377fe4e21030/packaging/flatpak/org.tenacity.Tenacity.json#L192 too?

**Comment by lmarz** [comment]
I think this is what you meant, right?

**Review by TheEvilSkeleton — DISMISSED** [review]
@lmarz yup, and I'm glad you noticed https://github.com/tenacityteam/tenacity/pull/642/files#diff-3aa1fa18fa1bebf563017c0fffdf4c058f97aa882b25e9403bc5db82aac1c048R212 because I forgot to link it.

**Review by n0toose — DISMISSED** [review]
Looks good to me as well.

File: packaging/flatpak/org.tenacity.Tenacity.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "command": "tenacity",
   "rename-desktop-file": "tenacity.desktop",
   "rename-icon": "tenacity",
-  "rename-appdata-file": "tenacity.metainfo.xml",
+  "rename-appdata-file": "org.tenacityaudio.Tenacity.metainfo.xml",
```
Comment (Be-ing) [inline packaging/flatpak/org.tenacity.Tenacity.json]: Is this line needed at all anymore?

File: packaging/flatpak/org.tenacity.Tenacity.json [hunk 2]
```diff
@@ -189,7 +189,7 @@
         "-DWX_CONFIG=/app/bin/wx-config"
       ],
       "post-install": [
-        "install -Dm644 ../help/tenacity.metainfo.xml -t /app/share/metainfo",
+        "install -Dm644 ../help/org.tenacityaudio.Tenacity.metainfo.xml -t /app/share/metainfo",
```
Comment (Be-ing) [inline packaging/flatpak/org.tenacity.Tenacity.json:191]: CMake should take care of this if it doesn't already.

**Comment by Be-ing** [comment]
Thanks for working on this.

File: packaging/flatpak/org.tenacity.Tenacity.json (hunk 1 above)
Comment (TheEvilSkeleton) [reply packaging/flatpak/org.tenacity.Tenacity.json]: Unneeded.

File: packaging/flatpak/org.tenacity.Tenacity.json (hunk 2 above)
Comment (lmarz) [reply packaging/flatpak/org.tenacity.Tenacity.json:191]: I think I need your or someones help with this. I don't really know CMake

**Review by nbsp — APPROVED** [review]
LGTM as well.

**Review by TheEvilSkeleton — APPROVED** [review]

File: packaging/flatpak/org.tenacity.Tenacity.json (hunk 2 above)
Comment (Be-ing) [reply packaging/flatpak/org.tenacity.Tenacity.json:191]: Okay, that can be done in another PR.

**Review by Be-ing — APPROVED** [review]

**Comment by n0toose** [comment]
Merged as it's been approved by two reviewers that are a part of our organization, while counting in @TheEvilSkeleton's review in the "time before the change should be merged" rule, as I believe that they are highly trustworthy as far as these sorts of changes are concerned and I originally tasked them with reviewing this proposed change in the first place.

**Comment by emabrey** [comment]
This broke flatpak packaging https://github.com/tenacityteam/tenacity/runs/3716847789?check_suite_focus=true

**Comment by emabrey** [comment]
I think line 7 of the flatpak manifest might be the problem:

```
 "rename-desktop-file": "tenacity.desktop",
 ```

**Comment by TheEvilSkeleton** [comment]
Thanks for pointing it out.

**Comment by Semisol** [comment]
I recommend that a new issue or PR is created for the tracking of the bug.

**Comment by TheEvilSkeleton** [comment]
@Semisol we already took care of it https://github.com/tenacityteam/tenacity/commit/fe5b64a1c2c1990b4c3f863948113e55503d73be.

**Comment by Semisol** [comment]
Okay, then we should be fine.

**Comment by n0toose** [comment]
> I think line 7 of the flatpak manifest might be the problem:
> 
> ```
>  "rename-desktop-file": "tenacity.desktop",
> ```

I seriously wonder how this actually went past us, all of the 11 checks seem to have passed at the time when I merged this!

Does this warrant further investigation?

**Comment by Semisol** [comment]
> I seriously wonder how this actually went past us, all of the 11 checks seem to have passed at the time when I merged this!

There was no Flatpak check running

**Comment by emabrey** [comment]
> 
> 
> > I think line 7 of the flatpak manifest might be the problem:
> > ```
> >  "rename-desktop-file": "tenacity.desktop",
> > ```
> 
> I seriously wonder how this actually went past us, all of the 11 checks seem to have passed at the time when I merged this!
> 
> Does this warrant further investigation?

The flatpak run takes so long that you have to manually activate it on a PR.

**Comment by n0toose** [comment]
> There was no Flatpak check running

> The flatpak run takes so long that you have to manually activate it on a PR.

Goddang it!

**Pull request merged**